pyi.Dockerfile

```diff
@@ -111,33 +111,72 @@ RUN ./node_builder_musl_env.sh
 COPY scripts/build_node_package.sh .
 RUN ./build_node_package.sh
 
-# bcc helpers
+# building bcc along with helpers
 # built on newer Ubuntu because they require new clang (newer than available in GPROFILER_BUILDER's CentOS 7)
 # these are only relevant for modern kernels, so there's no real reason to build them on CentOS 7 anyway.
-FROM ubuntu${PYPERF_BUILDER_UBUNTU} AS bcc-helpers
+FROM ubuntu${PYPERF_BUILDER_UBUNTU} AS bcc-build
 WORKDIR /tmp
 
 RUN if [ "$(uname -m)" = "aarch64" ]; then \
         exit 0; \
     fi && \
     apt-get update && \
-    apt-get install -y --no-install-recommends \
-        clang-10 \
-        libelf-dev \
-        make \
-        build-essential \
-        llvm \
-        ca-certificates \
-        git
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+      curl \
+      build-essential \
+      iperf llvm-12-dev \
+      clang-12 libclang-12-dev \
+      cmake \
+      python3 python3-pip \
+      flex \
+      libfl-dev \
+      bison \
+      libelf-dev \
+      libz-dev \
+      liblzma-dev \
+      ca-certificates \
+      git \
+      patchelf scons
 
 COPY --from=perf-builder /bpftool /bpftool
 
 COPY scripts/bcc_helpers_build.sh .
 RUN ./bcc_helpers_build.sh
 
+# build bcc
+# TODO: copied from the main Dockerfile... but modified a lot. we'd want to share it some day.
+WORKDIR /tmp
+
+COPY ./scripts/libunwind_build.sh .
+RUN if [ "$(uname -m)" = "aarch64" ]; then \
+      exit 0; \
+    fi && \
+    ./libunwind_build.sh
+
+WORKDIR /bcc
+COPY ./scripts/pyperf_build.sh .
+RUN ./pyperf_build.sh
 
-# bcc & gprofiler
+# build staticx dedicated for PyPerf and process PyPerf binary;
+# apply patch to ensure staticx bootloader propagates dump signal to actual PyPerf binary
+COPY scripts/staticx_for_pyperf_patch.diff staticx_for_pyperf_patch.diff
+# hadolint ignore=DL3003
+RUN if [ "$(uname -m)" = "aarch64" ]; then \
+      exit 0; \
+    fi && \
+    git clone -b v0.13.8 https://github.com/JonathonReinhart/staticx.git && \
```

Comment:
Any particular reason you upgraded to v0.13.8? Just for the sake of a new software, or was there a problem with v0.13.6?

Reply:
No, I just chose the recent one.
But to speed up the build I will change back to v0.13.6. Nothing changed in bootloader v0.13.2..v0.13.8.
👍 Will fix.